civis/io/_tables.py

```diff
@@ -172,17 +177,28 @@ def read_civis_sql(sql, database, use_pandas=False, job_name=None,
     """
     if use_pandas and NO_PANDAS:
         raise ImportError("use_pandas is True but pandas is not installed.")
-    with tempfile.NamedTemporaryFile(mode="w+") as f:
-        csv_poll = civis_to_csv(f.name, sql=sql, database=database,
-                                job_name=job_name, credential_id=credential_id,
-                                polling_interval=polling_interval,
-                                archive=archive, api_key=api_key)
-        csv_poll.result()
-        if use_pandas:
-            data = pd.read_csv(f.name, **kwargs)
-        else:
-            data = list(csv.reader(f, **kwargs))
+    client = APIClient(api_key=api_key)
+    script_id, run_id = _sql_script(client, sql, database,
+                                    job_name, credential_id)
+    poll = PollableResult(client.scripts.get_sql_runs,
+                          (script_id, run_id),
+                          polling_interval)
+    if archive:
 
+        def f(x):
+            return client.scripts.put_sql_archive(script_id, True)
+
+        poll.add_done_callback(f)
+    poll.result()
+    url = client.scripts.get_sql_runs(script_id, run_id)["output"][0]["path"]
```

Comment:
This line will raise an `IndexError` if the script doesn't return any rows. (`client.scripts.get_sql_runs(script_id, run_id)["output"]` will be an empty list). What about something like:
```
outputs = client.scripts.get_sql_runs(script_id, run_id)["output"]
if not outputs:
    raise EmptyResultError
url = outputs[0]["path"]
```